substrate/client/db/src/lib.rs

```diff
@@ -1707,7 +1708,8 @@ impl<Block: BlockT> Backend<Block> {
 							&(start, end).encode(),
 						);
 					}
-				} else if number > best_num + One::one() &&
+				} else if operation.create_gap &&
+					number > best_num + One::one() &&
```

Comment:
`number > best_num + One::one()` implies  `number > One::one()`, so we can eliminate the latter